Capture TablePathPrefix (and other parts of the parser context) in CREATE VIEW

jepett0 opened the pull request:
https://github.com/ydb-platform/ydb/issues/8992

We need to capture the parser context in order to use it later during a `SELECT` from the view to ensure that the results of the compilation of the view query text are the same as they were during `CREATE VIEW`. Postgres achieves the same by persisting AST of the view query, but our parser cannot handle this. More details on the problem can be found in the [RFC](https://github.com/ydb-platform/ydb-rfc/blob/main/view_context_capture.md).

jepett0 committed:
Capture TablePathPrefix (and other parts of the parser context) in CREATE VIEW

The captured context is used later to compile the view query in order to rewrite the read from the view node during a select from the view.

jepett0 committed:
fix broken unit tests

jepett0 commented on ydb/core/kqp/provider/yql_kikimr_datasink.cpp:
```diff
@@ -186,9 +186,9 @@ class TKiSinkIntentDeterminationTransformer: public TKiSinkVisitorTransformer {
-        return TStatus::Error;
+        Y_UNUSED(node);
+        Y_UNUSED(ctx);
+        return TStatus::Ok;
```
The same (i.e. nothing) is done in [HandleCreateObject](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/core/kqp/provider/yql_kikimr_datasink.cpp#L179) 12 lines above.

jepett0 commented on ydb/core/kqp/ut/view/view_ut.cpp:
```diff
@@ -408,6 +408,46 @@ Y_UNIT_TEST_SUITE(TSelectFromViewTest) {
         CompareResults(etalonResults, selectFromViewResults);
     }
 
+    Y_UNIT_TEST(OneTableUsingRelativeName) {
```
I would like to write an extensive test suite for views after this PR, so I add only a basic test of the new feature.

jepett0 commented on ydb/library/yql/sql/settings/protos/translation_settings.proto:
```diff
@@ -0,0 +1,14 @@
+
+option java_package = "com.yandex.yql.proto";
+
+message TTranslationSettings {
```
NB: The contents of this .proto are the most import to discuss in this PR.

### Why out of the all fields of the [TTranslationSettings](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.h#L73) I have decided to serialize only:
- PathPrefix
- SyntaxVersion
- AnsiLexer
- Antlr4Parser
- PgParser
- and pragmas?

#### Answer

The scope of the view context capturing in the first release of views is discussed a bit on the [wiki](https://wiki.yandex-team.ru/users/deminds/view-first-release-roadmap/#capture-from-the-very-beginning).

The main reason for choosing only these settings is that they seem to be the most important among those stored in the translation settings struct. Also they [have](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.cpp#L71) dedicated control comments, which somehow makes them more important in my mind.

I tried to implement the following approach:
- capture those fields of the TTranslationSettings struct that affect the compilation of the view query
- inherit those fields of the TTranslationSettings struct that are runtime-affecting from the `SELECT` from the view [context](https://github.com/ydb-platform/ydb-rfc/blob/main/view_context_capture.md#cause)

Here are some examples of the runtime-affecting parts of the TTranslationSettings struct:
- [DefaultCluster](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.h#L96)
  I inherit it from the enclosing parser context:
  - [here](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/library/yql/sql/v1/sql_translation.cpp#L66) for verification of the query during `CREATE VIEW` before storing it in the view
  - [here](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/core/kqp/host/kqp_translate.cpp#L93) during an actual [rewriting](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/core/kqp/provider/yql_kikimr_datasource.cpp#L777) of the read from the view node during `SELECT` from the view
- [ClusterMapping](https://github.com/ydb-platform/ydb/blob/main/ydb/library/yql/sql/settings/translation_settings.h#L78)
  I inherit it from the enclosing parser context:
  - [here](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/library/yql/sql/v1/sql_translation.cpp#L64) for verification of the query text during `CREATE VIEW`
  - [here](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/core/kqp/host/kqp_translate.cpp#L94) for the actual rewriting of a read from a view

Some translation settings are not being captured, because they are irrelevant for views:
- [V0Behavior](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L107C21-L107C31)
  We don't support V0 syntax in views.
- other V0 settings: V0ForceDisable, WarnOnV0, V0WarnAsError
- [DeclaredNamedExprs](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L93C32-L93C50)
  We plan to support named expressions inside view queries with a different approach. This feature is scheduled for the next view release.
- [EndOfQueryCommit](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L99)

Some translation settings are going to be captured in the next release:
- [GUC settings](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L123)
  You can read more on the first and the second releases features on [wiki](https://wiki.yandex-team.ru/users/deminds/view-first-release-roadmap/#view-context-capture-development-plan).
  
Other translation settings are not captured, because I don't know if they are needed for views:
- [Bindings](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L87)
- [IncrementCounter](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L97)
- ...

jepett0 commented on ydb/library/yql/sql/settings/translation_settings.h:
```diff
@@ -125,6 +126,9 @@ namespace NSQLTranslation {
         bool PgSortNulls = false;
+
+        NYql::NProto::TTranslationSettings Serialize() const;
+        void UpdateWith(const NYql::NProto::TTranslationSettings& serializedSettings);
```
Why `UpdateWith` and not something like `static TTranslationSettings Deserialize(const NYql::NProto::TTranslationSettings& serializedSettings)`?

It matches the way I intend to use serialized settings, i.e. overwrite some fields in the `TTranslationSettings` inherited from the enclosing context.

See usage [here](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/library/yql/sql/v1/sql_translation.cpp#L62) and [here](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/core/kqp/provider/rewrite_io_utils.cpp#L27).

jepett0 commented on ydb/library/yql/sql/settings/translation_settings.cpp:
```diff
@@ -67,6 +67,37 @@ namespace NSQLTranslation {
+
+        #undef DeserializeSetting
+
+        Flags.insert(serializedSettings.GetPragmas().begin(), serializedSettings.GetPragmas().end());
```
#### What if the `Flags` already contain [`AnsiCurrentRow`](https://ydb.tech/docs/en/yql/reference/syntax/pragma#ansicurrentrow) and `serializedSettings` are adding `DisableAnsiCurrentRow`?

See the relevant code [here](https://github.com/ydb-platform/ydb/blob/f38e45016276218330480a40a5575b3d5523522e/ydb/library/yql/sql/v1/context.cpp#L112). I believe, the last pragma will win, i.e. `DisableAnsiCurrentRow` will be in effect.

Takeaway: serialized settings will take precedence over those fields of the TTranslationSettings struct that they are updating.

jepett0 commented on ydb/library/yql/sql/v1/sql_query.cpp:
```diff
@@ -1239,8 +1239,12 @@ bool TSqlQuery::Statement(TVector<TNodePtr>& blocks, const TRule_sql_stmt_core&
             std::map<TString, TDeferredAtom> features;
-            ParseViewOptions(features, node.GetRule_with_table_settings4());
-            ParseViewQuery(features, node.GetRule_select_stmt6());
+            if (!ParseViewOptions(features, node.GetRule_with_table_settings4())) {
```
Here I fix a pretty impactful bug: the results of the view query verification by [building](https://github.com/ydb-platform/ydb/blob/f38e45016276218330480a40a5575b3d5523522e/ydb/library/yql/sql/v1/sql_translation.cpp#L4855) an AST of the query text were discarded previously. Early validation of the queries stored in views didn't work.

vitstn commented on ydb/library/yql/sql/settings/translation_settings.h:
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <ydb/library/yql/core/pg_settings/guc_settings.h>
+#include <ydb/library/yql/sql/settings/protos/translation_settings.pb.h>
```
не надо ссылаться на protobuf-ы в основной либе

сделайте отдельную library, в которой будет save/load в protobuf классы

и вообще этот protobuf и library унесите в ydb

jepett0 replied:
Не знаю, как унести в YDB, учитывая, что эта библиотека нужна в `sql_translation.cpp` для валидации текста запроса view.

В отдельную библиотеку выделил, но она по-прежнему лежит в YQL юрисдикции: `ydb/library/yql/sql/settings/serializer/serializer.h`.

jepett0 replied:
Унёс всё в YDB. Мотивация следующая:
- KQP [создаёт](https://github.com/ydb-platform/ydb/blob/bc6056258a1513f507c687bb6e56f0a0adf0fd6f/ydb/core/kqp/host/kqp_translate.cpp#L162) `TTransltaionSettings`, с которыми будет запущен парсер. Парсер не меняет эти настройки.
- Сериализация нужна в KQP в момент [создания](https://github.com/ydb-platform/ydb/blob/bc6056258a1513f507c687bb6e56f0a0adf0fd6f/ydb/core/kqp/gateway/behaviour/view/manager.cpp#L62) TModifyScheme для CreateView.
- Раз translation settings и создаются, и используются вне парсера, то логично их там и сериализовывать. Парсер в таком подходе ничего не знает об этой части сохранения контекста для view.

Важно отметить, что теперь, когда мы сериализуем контекст в KQP, этот процесс становится по смыслу равен сохранению применённых дефолтных настроек парсера в теле view. Query replay для тех же целей запоминает [query type](https://github.com/ydb-platform/ydb/blob/c10bc68a4c3b83327db1fe2ad59bdb345b5d1a23/ydb/core/kqp/provider/yql_kikimr_provider.h#L74), с которым был выполнен запрос. Мы могли бы делать так же (query type было бы вполне достаточно), но поскольку view - это объект из мира парсера, то мы выбрали формат более подходящий для объекта из парсера.

vitstn commented on ydb/library/yql/sql/v1/proto_parser/ya.make:
```diff
@@ -1,6 +1,7 @@
 LIBRARY()
 
 PEERDIR(
+    ydb/library/yql/sql/settings/protos
```
этой зависимости быть не должно

vitstn commented on ydb/library/yql/sql/v1/sql_translation.cpp:
```diff
@@ -54,17 +55,27 @@ TString CollectTokens(const TRule_select_stmt& selectStatement) {
 }
 
-NSQLTranslation::TTranslationSettings CreateViewTranslationSettings(const NSQLTranslation::TTranslationSettings& base) {
+NSQLTranslation::TTranslationSettings CreateViewTranslationSettings(
```
нужно извне принести интерфейс по загрузке/выгрузке translation settings.
сам sql парсер не должен зависеть от конкретной реализации

jepett0 replied:
Сделал интерфейс для `TTranslationSettingsSerializer` с использованием `TString` как формата сериализованных `TTranslationSettings`. Парсер не знает о том, что внутри для промежуточных построений этот сериализатор использует `NYql::NProto::TTranslationSettings`. Парсер работает только с `TTranslationSettings` и `TString`.

jepett0 replied:
Перенёс [сериализацию](https://github.com/ydb-platform/ydb/blob/cfbc22106c578d16267d682d1cd3fcd2b9dc6141/ydb/core/kqp/gateway/behaviour/view/manager.cpp#L64) `TTranslationSettings` целиком в KQP

jepett0 committed:
review fix 1

- Inject dependencies on translation settings serializer to TSqlTranslation class.
- Move translation settings serializer to a separate library.

jepett0 commented on ydb/library/yql/sql/settings/serializer/serializer.cpp:
```diff
@@ -0,0 +1,57 @@
+
+    #undef DeserializeSetting
+
+    settings.Flags.insert(serializedSettings.GetPragmas().begin(), serializedSettings.GetPragmas().end());
```
#### What if the `Flags` already contain [`AnsiCurrentRow`](https://ydb.tech/docs/en/yql/reference/syntax/pragma#ansicurrentrow) and `serializedSettings` are adding `DisableAnsiCurrentRow`?

See the relevant code [here](https://github.com/ydb-platform/ydb/blob/f38e45016276218330480a40a5575b3d5523522e/ydb/library/yql/sql/v1/context.cpp#L112). I believe, the last pragma will win, i.e. `DisableAnsiCurrentRow` will be in effect.

Takeaway: serialized settings will take precedence over those fields of the TTranslationSettings struct that they are updating.

jepett0 commented on ydb/library/yql/sql/settings/serializer/proto/translation_settings.proto:
```diff
@@ -0,0 +1,14 @@
+
+option java_package = "com.yandex.yql.proto";
+
+message TTranslationSettings {
```
NB: The contents of this .proto are the most import to discuss in this PR.

### Why out of the all fields of the [TTranslationSettings](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.h#L73) I have decided to serialize only:
- PathPrefix
- SyntaxVersion
- AnsiLexer
- Antlr4Parser
- PgParser
- and pragmas?

#### Answer

The scope of the view context capturing in the first release of views is discussed a bit on the [wiki](https://wiki.yandex-team.ru/users/deminds/view-first-release-roadmap/#capture-from-the-very-beginning).

The main reason for choosing only these settings is that they seem to be the most important among those stored in the translation settings struct. Also they [have](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.cpp#L71) dedicated control comments, which somehow makes them more important in my mind.

I tried to implement the following approach:
- capture those fields of the TTranslationSettings struct that affect the compilation of the view query
- inherit those fields of the TTranslationSettings struct that are runtime-affecting from the `SELECT` from the view [context](https://github.com/ydb-platform/ydb-rfc/blob/main/view_context_capture.md#cause)

Here are some examples of the runtime-affecting parts of the TTranslationSettings struct:
- [DefaultCluster](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.h#L96)
  I inherit it from the enclosing parser context:
  - [here](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/library/yql/sql/v1/sql_translation.cpp#L66) for verification of the query during `CREATE VIEW` before storing it in the view
  - [here](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/core/kqp/host/kqp_translate.cpp#L93) during an actual [rewriting](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/core/kqp/provider/yql_kikimr_datasource.cpp#L777) of the read from the view node during `SELECT` from the view
- [ClusterMapping](https://github.com/ydb-platform/ydb/blob/main/ydb/library/yql/sql/settings/translation_settings.h#L78)
  I inherit it from the enclosing parser context:
  - [here](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/library/yql/sql/v1/sql_translation.cpp#L64) for verification of the query text during `CREATE VIEW`
  - [here](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/core/kqp/host/kqp_translate.cpp#L94) for the actual rewriting of a read from a view

Some translation settings are not being captured, because they are irrelevant for views:
- [V0Behavior](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L107C21-L107C31)
  We don't support V0 syntax in views.
- other V0 settings: V0ForceDisable, WarnOnV0, V0WarnAsError
- [DeclaredNamedExprs](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L93C32-L93C50)
  We plan to support named expressions inside view queries with a different approach. This feature is scheduled for the next view release.
- [EndOfQueryCommit](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L99)

Some translation settings are going to be captured in the next release:
- [GUC settings](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L123)
  You can read more on the first and the second releases features on [wiki](https://wiki.yandex-team.ru/users/deminds/view-first-release-roadmap/#view-context-capture-development-plan).
  
Other translation settings are not captured, because I don't know if they are needed for views:
- [Bindings](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L87)
- [IncrementCounter](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L97)
- ...

jepett0 committed:
minor fixes

jepett0 committed:
Major view context capture change

- capture TTranslationSettings in KQP
- runtime TContext parts as a view query text

jepett0 committed:
Move serializer to KQP

+ capture USE. It cannot be used in YDB, however, it is needed to past a sql unit test (v1/sql_ut.cpp).
+ some other minor changes

jepett0 commented on ydb/core/protos/yql_translation_settings.proto:
```diff
@@ -0,0 +1,15 @@
+
+option java_package = "ru.yandex.kikimr.proto";
+
+message TTranslationSettings {
```
NB: The contents of this .proto are the most import changes to discuss in this PR.

### Why out of the all fields of the [TTranslationSettings](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.h#L73) I have decided to serialize only:
- PathPrefix
- SyntaxVersion
- AnsiLexer
- Antlr4Parser
- PgParser
- and pragmas?

#### Answer

The scope of the view context capturing in the first release of views is discussed a bit on the [wiki](https://wiki.yandex-team.ru/users/deminds/view-first-release-roadmap/#capture-from-the-very-beginning).

The main reason for choosing only these settings is that they seem to be the most important among those stored in the translation settings struct. Also they [have](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.cpp#L71) dedicated control comments, which somehow makes them more important in my mind.

I tried to implement the following approach:
- capture those fields of the TTranslationSettings struct that affect the compilation of the view query
- inherit those fields of the TTranslationSettings struct that are runtime-affecting from the `SELECT` from the view [context](https://github.com/ydb-platform/ydb-rfc/blob/main/view_context_capture.md#cause)

Here are some examples of the runtime-affecting parts of the TTranslationSettings struct:
- [DefaultCluster](https://github.com/ydb-platform/ydb/blob/f3a1a6271b5375688cb0608efd644334cc28bbc9/ydb/library/yql/sql/settings/translation_settings.h#L96)
  I inherit it from the enclosing parser context:
  - [here](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/library/yql/sql/v1/sql_translation.cpp#L66) for verification of the query during `CREATE VIEW` before storing it in the view
  - [here](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/core/kqp/host/kqp_translate.cpp#L93) during an actual [rewriting](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/core/kqp/provider/yql_kikimr_datasource.cpp#L777) of the read from the view node during `SELECT` from the view
- [ClusterMapping](https://github.com/ydb-platform/ydb/blob/main/ydb/library/yql/sql/settings/translation_settings.h#L78)
  I inherit it from the enclosing parser context:
  - [here](https://github.com/ydb-platform/ydb/blob/414cd61a450b0e82a9196e7c0dd20fcc555055d5/ydb/library/yql/sql/v1/sql_translation.cpp#L64) for verification of the query text during `CREATE VIEW`
  - [here](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/core/kqp/host/kqp_translate.cpp#L94) for the actual rewriting of a read from a view

Some translation settings are not being captured, because they are irrelevant for views:
- [V0Behavior](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L107C21-L107C31)
  We don't support V0 syntax in views.
- other V0 settings: V0ForceDisable, WarnOnV0, V0WarnAsError
- [DeclaredNamedExprs](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L93C32-L93C50)
  We plan to support named expressions inside view queries with a different approach. This feature is scheduled for the next view release.
- [EndOfQueryCommit](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L99)

Some translation settings are going to be captured in the next release:
- [GUC settings](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L123)
  You can read more on the first and the second releases features on [wiki](https://wiki.yandex-team.ru/users/deminds/view-first-release-roadmap/#view-context-capture-development-plan).
  
Other translation settings are not captured, because I don't know if they are needed for views:
- [Bindings](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L87)
- [IncrementCounter](https://github.com/ydb-platform/ydb/blob/17a9a618a506cac75f5f7f64ed6df7850e1a664f/ydb/library/yql/sql/settings/translation_settings.h#L97)
- ...

vitstn commented on ydb/library/yql/sql/sql.h:
```diff
@@ -17,6 +17,8 @@ namespace NSQLTranslation {
 
     NYql::TAstParseResult SqlToYql(const TString& query, const TTranslationSettings& settings,
         NYql::TWarningRules* warningRules = nullptr, NYql::TStmtParseInfo* stmtParseInfo = nullptr);
+    NYql::TAstParseResult SqlToYql(const TString& query, TTranslationSettings& settings,
```
лучше сделать TTranslationSettings* effectiveSettings = nullptr параметр, куда будет записана то, на чем было выполнение.

если этот параметр задан, заводить на стеке TTranslationSettings перед мутацией не нужно

jepett0 committed:
Review fix

Remove duplicate SqlToYql function

jepett0 committed:
Remove ANTLR parser version persisting in views
